internal/rbd/nodeserver.go

```diff
@@ -1314,6 +1317,55 @@ func (ns *NodeServer) xfsSupportsReflink() bool {
 	return false
 }
 
+// ext4SupportsPrezeroed checks if the ext4 filesystem supports the
+// "assume_storage_prezeroed" option. It does this by creating a temporary
+// image file, attempting to format it with the ext4 filesystem using the
+// "assume_storage_prezeroed" option, and checking the output for errors.
+func (ns *NodeServer) ext4SupportsPrezeroed() bool {
+	if ext4HasPrezeroedSupport != ext4PrezeroedUnset {
+		return ext4HasPrezeroedSupport == ext4PrezeroedSupport
+	}
+
+	ctx := context.TODO()
+	tempImgFile, err := os.CreateTemp(os.TempDir(), "prezeroed.img")
+	if err != nil {
+		log.WarningLog(ctx, "failed to create temporary image file: %v", err)
+
+		return false
+	}
+	defer os.Remove(tempImgFile.Name())
+
+	if err = createSparseFile(tempImgFile, 1); err != nil {
+		log.WarningLog(ctx, "failed to create sparse file: %v", err)
+
+		return false
+	}
+
+	diskMounter := &mount.SafeFormatAndMount{Interface: ns.Mounter, Exec: utilexec.New()}
+
+	// `-n` Causes mke2fs to not actually create a file system
+	// but display what it would do if it were to create a file system.
+	out, err := diskMounter.Exec.Command(
+		"mkfs.ext4",
+		"-n",
+		"-Enodiscard,assume_storage_prezeroed=1",
```

Comment:
I think it is nice to have the options mentioned as a const somewhere, they are here, but also in `getMkfsArgs()`.

Maybe keep the default map, and use `ext4.prezeroed` as fstype to get the options?

Reply:
> I think it is nice to have the options mentioned as a const somewhere, they are here

I don't have an opinion about it being expressed as a const, but I think it should be a single option.  This function is supposed to be checking whether ext4 supports "prezeroed", so it should be just that -- pass only `assume_storage_prezeroed=1`, `nodiscard` isn't relevant here.

Reply:
Removed the `nodiscard` flag.